File: I.py
n, m = map(int, input().split())
rows = [input() for _ in range(n)]
cols = [''.join(tup) for tup in zip(*rows)] # transpose rows to get cols

def seq(s):
    lens = [x for x in map(len, s.split('#')) if x != 0]
    return min(lens) if lens else 10**9

ans = min(*(seq(s) for s in rows), *(seq(s) for s in cols))
print(ans)

File: I.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
  cin.tie(0)->sync_with_stdio(0);
  cin.exceptions(cin.failbit);

  int n, m;
  cin >> n >> m;

  vector<string> rows(n), cols(m);
  for (int i = 0; i < n; ++i) cin >> rows[i];
  for (int i = 0; i < n; ++i) for (int j = 0; j < m; ++j) {
    cols[j].push_back(rows[i][j]);
  }

  auto seq = [&](string &s) {
    int cnt = 0, ans = 1e9;
    for (int i = 0; i < int(s.size()); ++i) {
      if (s[i] == '.') {
        ++cnt;
      } else {
        if (cnt) ans = min(ans, cnt);
        cnt = 0;
      }
    }
    if (cnt) ans = min(ans, cnt);
    return ans;
  };

  int ans = 1e9;
  for (auto s: rows) ans = min(ans, seq(s));
  for (auto s: cols) ans = min(ans, seq(s));
  cout << ans << '\n';
}
